### PR TITLE
v2.3.1

### PR DESCRIPTION
**Pull request by mtakla16**
* Add `Triniti\AppleNews\Component\ConditionalComponent` to enable conditional properties for components

### CODE REVIEW

File: src/AppleNews/Component/Component.php [hunk 1]
```diff
@@ -181,4 +187,73 @@ public function setStyle($style = null): self
...
+    /**
+     * @param ConditionalComponent[] $conditionals
+     *
+     * @return static
+     */
+    public function setConditional(?array $conditionals = []): self
+    {
+        $this->conditional = [];
+
+        if (null !== $conditionals) {
```
Comment (bhsiao55) [inline src/AppleNews/Component/Component.php]: maybe use !empty($conditional) instead so it checks for empty array and null

File: src/AppleNews/Component/Component.php [hunk 2]
```diff
@@ -181,4 +187,73 @@ public function setStyle($style = null): self
...
+    }
+
+    /**
+     * @param ConditionalComponent[] $conditionals
+     *
+     * @return static
+     */
+    public function addConditionals(?array $conditionals = []): self
+    {
+        if (null !== $conditionals) {
```
Comment (bhsiao55) [inline src/AppleNews/Component/Component.php]: maybe use !empty($conditional) here too

File: src/AppleNews/Component/ConditionalComponent.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+namespace Triniti\AppleNews\Component;
+
+use Assert\Assertion;
+use Triniti\AppleNews\Condition;
+
+/**
+ * @link https://developer.apple.com/documentation/apple_news/conditionalcomponent
+ */
+class ConditionalComponent extends Component
```
Comment (bhsiao55) [inline src/AppleNews/Component/ConditionalComponent.php]: extending Component might not work here.  I see properties on Component that shouldn't be included on ConditionalComponent

**Review by bhsiao55 — CHANGES_REQUESTED** [review]

File: src/AppleNews/Component/ConditionalComponent.php (hunk 3 above)
Comment (mtakla16) [reply src/AppleNews/Component/ConditionalComponent.php]: Gotcha, I've updated it to extend AppleNewsObject instead

**Review by gdbrown — APPROVED** [review]

**Pull request merged**